### ISSUE TITLE
x/net/websocket: Integer overflow on hybiFrameHeader Length

### ISSUE THREAD

**Issue by robertmd** [golang/go#11999]
Currently on go1.4.2 windows/amd64, hybiFrameHeader uses int64 for the length of the packet. If a header size is greater than int64 the length will become negative. This causes the data in the packet to become the header of the next packet.

Per the RFC specs the length should be uint64 or the websocket library should check for overflows.

**Comment by mikioh** [comment]
According to https://tools.ietf.org/html/rfc6455,
```
Payload length:  7 bits, 7+16 bits, or 7+64 bits

      The length of the "Payload data", in bytes: ... (snip)
      following 8 bytes interpreted as a 64-bit unsigned integer (the
      most significant bit MUST be 0) are the payload length. 
```

**Comment by mikioh** [comment]
The current implementation is fine because a) it validates the MSB of payload length, b) it doesn't support long payload frames. Merged into #2134.

**Comment by mikioh** [comment]
Ah, I missed that there's no MSB check for extended payload length. Thanks.
